spec/lib/msf/core/exploit/remote/remote/http/http_cookie_jar_spec.rb

```diff
@@ -0,0 +1,278 @@
+require 'spec_helper'
+require 'msf/core/exploit/remote/http/http_cookie_jar'
+require 'msf/core/exploit/remote/http/http_cookie'
+
+RSpec.describe Msf::Exploit::Remote::HTTP::HttpCookieJar do
+  def random_string(min_len = 1, max_len = 12)
+    str = Faker::Alphanumeric.alpha(number: max_len)
+    str[0, rand(min_len..max_len)]
+  end
+
+  def cookie
+    Msf::Exploit::Remote::HTTP::HttpCookie.new(
+      random_string,
+      random_string,
+      max_age: rand(1..100),
+      path: '/' + random_string,
+      secure: rand(0..1) == 1,
+      httponly: rand(0..1) == 1,
+      domain: random_string
+    )
+  end
+
+  let(:cookie_jar) { described_class.new }
+
+  before(:each) do
+    Timecop.freeze(Time.local(2008, 9, 5, 10, 5, 30))
+  end
+
+  after(:each) do
+    Timecop.return
+  end
+
+  describe 'empty?' do
+    it 'will return true when no cookies are in a cookie_jar' do
+      # cookie_jar made in before
+
+      e = cookie_jar.empty?
+
+      expect(e).to eq(true)
+    end
+
+    it 'will return false when a cookie has been added to a cookie_jar' do
+      c = cookie
+
+      cookie_jar.add(c)
+      e = cookie_jar.empty?
+
+      expect(e).to eq(false)
+    end
+  end
+
+  describe 'clear' do
+    it 'will make no changes to an empty cookiejar' do
+      # empty cookie_jar made in before
+
+      cookie_jar.clear
+
+      expect(cookie_jar.empty?).to eq(true)
+    end
+
+    it 'will return false when a cookie has been added to a cookie_jar' do
+      c = cookie
+
+      cookie_jar.add(c)
+      cookie_jar.clear
+
+      expect(cookie_jar.empty?).to eq(true)
+    end
```

Comment:
`will return false` but you have `eq(true)` one of these has gotta be wrong :)

Reply:
Yeah that's a copypasta typo in the test description 👍